include/deal.II/base/table.h

```diff
@@ -2367,6 +2624,170 @@ Table<2,T>::n_cols () const
 
 
 
+//---------------------------------------------------------------------------
+namespace TransposeTableIterators
+{
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase()
+    :
+    object(nullptr),
+    linear_index(std::numeric_limits<decltype(linear_index)>::max())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const object_pointer_type table)
+    :
+    object(table),
+    // TODO why can't this be object->values.size()?
+    linear_index(table->values.size())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const AccessorBase<T, false> &a)
+    :
+    object(a.object),
+    linear_index(a.linear_index)
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const object_pointer_type table,
+                                           const std::ptrdiff_t index)
+    :
+    object(table),
+    linear_index(index)
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  const T &
+  AccessorBase<T, Constness>::value() const
+  {
+    Assert(0 <= this->linear_index,
+           ExcMessage("The current iterator points outside of the table."));
+    return this->object->values[this->linear_index];
+  }
+
+
+
+  template <typename T, bool Constness>
+  inline
+  typename AccessorBase<T, Constness>::size_type
+  AccessorBase<T, Constness>::AccessorBase::row() const
+  {
+    Assert(!object->empty(), ExcMessage("An empty table has no rows or columns."));
+    return linear_index % object->n_rows();
+  }
+
+
+
+  template <typename T, bool Constness>
+  inline
+  typename AccessorBase<T, Constness>::size_type
+  AccessorBase<T, Constness>::AccessorBase::column() const
+  {
+    Assert(!object->empty(), ExcMessage("An empty table has no rows or columns."));
+    return linear_index/object->n_rows();
+  }
+
+
+  template <typename T>
+  inline
+  T &
+  Accessor<T, false>::operator = (const T &t) const
```

Comment:
I should have mentioned this at the class declaration: The common return type of `operator=` is the lhs operand, i.e., `*this`. This would also make the implementation of this function simpler.

Same for the `const`-version below.

Reply:
Fixed.